coderd/database/dbauthz/dbauthz.go

```diff
@@ -5507,6 +5507,32 @@ func (q *querier) UpdateWorkspaceAgentLogOverflowByID(ctx context.Context, arg d
 }
 
 func (q *querier) UpdateWorkspaceAgentMetadata(ctx context.Context, arg database.UpdateWorkspaceAgentMetadataParams) error {
+	// Fast path: Check if we have an RBAC object in context.
+	// This is set by the workspace agent RPC handler to avoid the expensive
+	// GetWorkspaceByAgentID query for every metadata update.
+	//
+	// NOTE: The cached RBAC object is refreshed every 5 minutes. After a prebuild
+	// claim, there may be up to a 5-minute window where this uses stale owner_id.
+	if rbacObj, ok := WorkspaceRBACFromContext(ctx); ok {
+		// Verify the RBAC object is valid (has required fields).
+		if !rbacObj.IsEmpty() && rbacObj.Type == rbac.ResourceWorkspace.Type {
```

Comment:
`WorkspaceRBACFromContext` should never return an invalid `RBACObject` imo. The `WithWorkspaceRBAC` protects against these errors like the wrong type.

So we can trust it and not guard against things if `ok == true`